packages/medusa/src/api/routes/store/carts/get-cart.ts

```diff
@@ -69,7 +76,27 @@ export default async (req, res) => {
       })
     }
   }
+  const shouldSetAvailability = req.retrieveConfig.relations?.some((rel) =>
+    rel.includes("variant")
+  )
+
+  const select = [...req.retrieveConfig.select]
+  const salesChannelsEnabled = featureFlagRouter.isFeatureEnabled(
+    SalesChannelFeatureFlag.key
+  )
+
+  if (salesChannelsEnabled) {
+    select.push("sales_channel_id")
+  }
 
   const data = await cartService.retrieveWithTotals(id, req.retrieveConfig)
+
+  if (shouldSetAvailability && salesChannelsEnabled) {
```

Comment:
question: Do we need to add `&& salesChannelsEnabled` to the store-facing product endpoints?